packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -663,6 +663,42 @@ InternalModel.prototype = {
     this._inFlightAttributes = Ember.create(null);
   },
 
+    /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
+    var changedKeys = [];
+
+    if (updates && typeof updates === 'object') {
+      var original, i, value, key;
+      var keys = Ember.keys(updates);
+      var length = keys.length;
+
+      original = merge({}, this._data);
+      original = merge(original, this._inFlightAttributes);
+
+      for (i = 0; i < length; i++) {
+        key = keys[i];
+        value = updates[key];
+
+        // A value in _attributes means the user has a local change to
+        // this attributes. We never override this value when merging
+        // updates from the backend so we should not sent a change
+        // notification if the server value differs from the original.
+        if (this._attributes[key] !== undefined) {
+          continue;
+        }
+
+        if (original[key] !== value) {
```

Comment:
Can we switch this to use Ember.isEqual while we're at it? http://emberjs.com/api/#method_isEqual